### ISSUE TITLE
Cannot use 'in' operator to search for  '^2.1.0' in undefined

### ISSUE THREAD

**Issue by danilovaz** [yarnpkg/yarn#953]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
```bash
$ time yarn
yarn install v0.15.1
info No lockfile found.
[1/4] Resolving packages...
error TypeError: Cannot use 'in' operator to search for '^2.1.0' in undefined
    at /home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:62:24
    at next (native)
    at step (/home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
    at Promise.F (/home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/node_modules/core-js/library/modules/_export.js:35:28)
    at /home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
    at Function.findVersionInRegistryResponse (/home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:72:7)
    at /home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:89:34
    at next (native)
    at step (/home/danilovaz/.nvm/versions/node/v6.7.0/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
info Visit http://yarnpkg.com/en/docs/cli/install for documentation about this command.
yarn  1,57s user 0,13s system 23% cpu 7,294 total
```

**If the current behavior is a bug, please provide the steps to reproduce.**
I just do: `time yarn`

I'm using *Ember* and my package.json is default:
```json
{
  "name": "myProject",
  "version": "0.6.3",
  "description": "",
  "private": true,
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build",
    "start": "ember server",
    "test": "ember test"
  },
  "repository": "",
  "engines": {
    "node": ">= 0.10.0"
  },
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "autoprefixer": "^6.3.6",
    "broccoli-asset-rev": "^2.2.0",
    "broccoli-clean-css": "^1.1.0",
    "ember-ajax": "0.7.1",
    "ember-cli": "2.6.0",
    "ember-cli-app-version": "^1.0.0",
    "ember-cli-babel": "^5.1.5",
    "ember-cli-base64-converter": "0.1.3",
    "ember-cli-dependency-checker": "^1.2.0",
    "ember-cli-facebook-js-sdk": "1.0.3",
    "ember-cli-htmlbars": "^1.0.1",
    "ember-cli-htmlbars-inline-precompile": "^0.3.1",
    "ember-cli-inject-live-reload": "^1.3.1",
    "ember-cli-mirage": "0.2.0",
    "ember-cli-moment-shim": "2.0.0",
    "ember-cli-qunit": "^1.2.1",
    "ember-cli-release": "0.2.8",
    "ember-cli-sri": "^2.0.0",
    "ember-cli-stylus": "^1.0.6",
    "ember-cli-uglify": "^1.2.0",
    "ember-data": "^2.6.0",
    "ember-disable-proxy-controllers": "^1.0.1",
    "ember-dragula": "1.9.3",
    "ember-export-application-global": "^1.0.4",
    "ember-load-initializers": "^0.5.0",
    "ember-moment": "7.0.0-beta.3",
    "ember-perfectscroll": "^0.1.0",
    "ember-power-select": "1.0.0-alpha.1",
    "ember-radio-button": "1.0.7",
    "ember-resolver": "^2.0.3",
    "ember-simple-auth": "1.1.0",
    "ember-websockets": "3.0.0",
    "emberx-file-input": "1.1.0",
    "gulp": "^3.9.1",
    "gulp-concat": "^2.6.0",
    "gulp-cssnano": "^2.1.2",
    "gulp-group-css-media-queries": "^1.1.0",
    "gulp-postcss": "^6.1.0",
    "gulp-sourcemaps": "^1.6.0",
    "liquid-fire": "0.23.1",
    "liquid-tether": "1.1.1",
    "loader.js": "^4.0.0",
    "lost": "^6.8.0",
    "phantomjs-prebuilt": "^2.1.6",
    "postcss-font-awesome": "^0.2.1",
    "poststylus": "^0.2.3",
    "rupture": "^0.6.1"
  },
  "dependencies": {
    "ember-cli-stylus": "^1.0.6",
    "rucksack-css": "^0.8.6"
  }
}
```

**What is the expected behavior?**
Get the all packages in my package.json installed.

**Please mention your node.js, yarn and operating system version.**
`nodejs: v6.7.0`
`yarn: 0.15.1`
`OS: Ubuntu 16.04.1 LTS`

**Comment by suitupalex** [comment]
Got this earlier when simply running `yarn`. The WiFi I was using was inconsistent so I think it had something to do with that. Running `yarn` immediately afterward did not reproduce the error.

**Comment by danilovaz** [comment]
@suitupalex You're right! That's so weird output error, anyway

**Comment by suitupalex** [comment]
Indeed. I wonder if this issue is worth keeping open to possibly figure out a better way to capture the error?

**Comment by dmorneau** [comment]
I also hit this exact same stack trace, and it worked fine on retry. My connection is pretty solid though.

**Comment by danilovaz** [comment]
Yeah, makes sense. I will open again, because my connection is pretty solid too.

**Comment by edba** [comment]
These seems to be the same issue I run into in #945 the RequestManager isn't able to handle a gzip request response.

**Comment by laurentpayot** [comment]
Same issue here with
```
nodejs: v6.8.0
yarn: v0.15.0
OS: Ubuntu 16.10 Desktop
```

**Comment by calebmer** [comment]
I keep getting this too except each time I rerun I’m getting different version numbers (for example `^6.5.0` the first time, `^1.1.0` the next time, etc.). After running it about 10 times packages resolved but I got an error about a corrupt tar file.

```
Node.JS: v6.7.0
yarn: v0.15.1
OS: MacOS Sierra
```

**Comment by wyze** [comment]
Solved by #1356.
